src/radical/pilot/raptor/worker_mpi.py

```diff
@@ -620,13 +620,21 @@ def _dispatch_function(self, task):
         # check if we have a serialized object
         self._log.debug('func serialized: %d: %s', len(func), func)
         try:
-            # FIXME: can we have a better test than try/except?  This hides
-            #        potential errors...
-            # FIXME: ensure we did not get args and kwargs from above
-            to_call, args, kwargs = PythonTask.get_func_attr(func)
+            to_call, _args, _kwargs = PythonTask.get_func_attr(func)
+            assert callable(to_call)
             py_func = True
-        except:
-            pass
+        except (ValueError, TypeError) as e:
```

Comment:
It is extremely important for debugging to know why `get_func_attr` was not successful. It is not clear to me under what circumstances, if any, you want `get_func_attr` to throw versus returning `None`, and what the raised exceptions should represent.

It is much less necessary to distinguish the reason for failure in this particular function, though, if a pickled function were the _last_ option checked for instead of the first. I would think this would also be the most expensive step. Shouldn't this be moved to the end? Then, there is no need to distinghish one type of failure from another. We can just log the exception, and raise the `ValueError` for "callable not found".

Reply:
done